libraries/standard/http/src/private/http_client_internal.h

```diff
@@ -1,6 +1,16 @@
 #ifndef HTTP_CLIENT_INTERNAL_H_
 #define HTTP_CLIENT_INTERNAL_H_
 
+#include "config.h"
+
+#if defined( HTTP_ENABLE_ASSERT ) && ( HTTP_ENABLE_ASSERT == 1 )
```

Comment:
I think we shouldn't have this and instead if customers are going to compile out asserts they will use the ANSI C definition NDEBUG in their compiler level macros.

Reply:
We discussed offline, and decided to defer the decision on supporting configurable asserts per library in consultation with the larger team.